shardingsphere-infra/shardingsphere-infra-rewrite/shardingsphere-infra-rewrite-engine/src/main/java/org/apache/shardingsphere/infra/rewrite/context/SQLRewriteContext.java

```diff
@@ -80,5 +80,8 @@ public void addSQLTokenGenerators(final Collection<SQLTokenGenerator> sqlTokenGe
      */
     public void generateSQLTokens() {
         sqlTokens.addAll(sqlTokenGenerators.generateSQLTokens(sqlStatementContext, parameters, schemaMetaData));
+        if (sqlStatementContext instanceof InsertStatementContext && null != ((InsertStatementContext) sqlStatementContext).getInsertSelectContext()) {
```

Comment:
It is indeed an easy way to handle the `select statement` in the `insert statement`. However,  two points get my concerns,

1. IMO, `SQLRewriteContext` does not need to know which `sqlstatment` it is and whether it is required to generate SQL tokens. So maybe I know `isGenerateSQLToken(sqlStatementContext)` of each `tokenGenerator` is a more proper place to give some similar estimations. However, this way is a bit hard to do the same thing.
2. If we call `sqlTokenGenerators.generateSQLTokens()` twice,  that means it runs `for each` twice internally. It is not efficient from my point.

I'd like to listen to your idea about it. :-)

Reply:
@tristaZero Modifying the `isGenerateSQLToken` and `generateSQLTokens` methods in each `tokenGenerator` is a bit difficult, but it seems to be a better solution, I will try it. 🙂